README.md

```diff
@@ -1,9 +1,205 @@
+The prefix option specifies that the fabtests files should be installed into
+the directory named `<directory>`. The executables will be located at
+`<directory>/bin`.
+
+```
+--with-valgrind=<directory>
+```
+
+Directory where valgrind is installed.If valgrind is found, then valgrind
+annotations are enabled. This may incur a performance penalty.
+
+```
+--enable-debug
+```
+
+Enable debug code paths. This enables various extra checks and allows for using
+the highest verbosity logging output that is normally compiled out in
+production builds.
+
+```
+--enable-<provider>=[yes|no|auto|dl|PATH]
+--disable-<provider>
+```
+
+This enables or disables the provider named `<provider>`. Each provider is
+enabled by default. The `dl` option builds the provider as a loadable library
+and the `PATH` option enables the provider and uses the installation given in
+`PATH`.
+
+### Examples
+
+Consider the following example:
+
+```
+$ ./configure --prefix=/opt/libfabric --disable-sockets && make -j 32 && sudo make install
+```
+This will tell Libfabric to disable the `sockets` provider, and install
+Libfabric in the `/opt/libfabric` tree. All other providers will be enabled if
+possible and all debug features will be disabled.
+
+Alternatively:
+
+```
+$ ./configure --prefix=/opt/libfabric --enable-debug --enable-psm=dl && make -j 32 && sudo make install
+```
+
+This will tell Libfabric to enable the `psm` provider as a loadable library,
+enable all debug code paths, and install Libfabric to the `/opt/libfabric`
+tree. All other providers will be enabled if possible.
+
+
+## Providers
+
+### Sockets
+
+***
+
+The `sockets` provider is a general purpose provider that can be used on any
+system that supports TCP sockets.  The provider is not intended to provide
+performance improvements over regular TCP sockets, but rather to allow
+developers to write, test, and debug application code even on platforms
+that do not have high-performance fabric hardware.  The sockets provider
+supports all libfabric provider requirements and interfaces.
+
+See the `fi_sockets(7)` man page for more details.
+
+### usNIC
+
+***
+
+The `usNIC` provider is designed to run over the Cisco VIC (virtualized NIC)
+hardware on Cisco UCS servers. It utilizes the Cisco usNIC (userspace NIC)
+capabilities of the VIC to enable ultra low latency and other offload
+capabilites on Ethernet networks.
+
+See the `fi_usnic(7)` man page for more details.
+
+#### Dependencies
+
+- The `usNIC` provider depends on having either `libnl1` or `libnl3`. If the
+  provider is being built from source, then the `libnl` header files will need
+  to be provided (e.g. install the "--devel" versions of the package).
+
+#### Configure options
+
+```
+--with-libnl=<directory>
+```
+
+If specified, look for libnl support. If it is not found then the `usNIC`
+provider will not be built. If `<directory>` is specified, then check in the
+directory and check for `libnl3`. If `libnl3` is not found, then check for
```

Comment:
check for `libnl` version 3.  If version 3 is not found, then check for version 1.